File: src/clase-06/python/strategy/test_strategy.py
from python.strategy.strategy import Navigator, CarRoute, WalkRoute, PublicTransportRoute

def test_strategy_switch():
    nav = Navigator(CarRoute())
    r1 = nav.build_route("A", "B")
    assert "auto" in r1

    nav.set_strategy(WalkRoute())
    r2 = nav.build_route("A", "B")
    assert "pie" in r2

    nav.set_strategy(PublicTransportRoute())
    r3 = nav.build_route("A", "B")
    assert "transporte público" in r3

File: src/clase-06/typescript/strategy/test_strategy.ts
import { Navigator, CarRoute, WalkRoute, PublicTransportRoute } from './strategy';

const nav = new Navigator(new CarRoute());
const r1 = nav.buildRoute('A','B');
if (!r1.includes('auto')) throw new Error('CarRoute falló');

nav.setStrategy(new WalkRoute());
const r2 = nav.buildRoute('A','B');
if (!r2.includes('pie')) throw new Error('WalkRoute falló');

nav.setStrategy(new PublicTransportRoute());
const r3 = nav.buildRoute('A','B');
if (!r3.includes('transporte público')) throw new Error('PublicTransportRoute falló');

console.log('test_strategy OK');
